Core/StitchCoreSDK/Sources/StitchCoreSDK/Auth/Internal/CoreStitchAuth+AuthStorage.swift

```diff
@@ -65,82 +63,67 @@ extension CoreStitchAuth {
     /**
      * Writes a list of `AuthInfo` structs into some underlying storage.
      *
-     * - parameters:
-     *     - currentUsersAuthInfo: An array of `AuthInfo` structs to write to `Storage`
-     *     - toStorage: The `Storage` to which to write the `AuthInfo`.
      * - throws: if the `AuthInfo` could not be encoded into JSON.
      */
-    internal func writeCurrentUsersAuthInfoToStorage(currentUsersAuthInfo currentUsers: [AuthInfo],
-                                                     toStorage storage: Storage) throws {
+    internal func writeCurrentUsersAuthInfoToStorage() throws {
         var currentUsersStore: [StoreAuthInfo] = []
-        for user in currentUsers {
+        for user in allUsersAuthInfo {
             currentUsersStore.append(StoreAuthInfo.init(withAuthInfo: user))
         }
 
-        storage.set(try JSONEncoder().encode(currentUsersStore), forKey: allCurrentUsersStorageKey)
+        do {
+            storage.set(try JSONEncoder().encode(currentUsersStore), forKey: allCurrentUsersStorageKey)
+        } catch {
+            throw StitchError.clientError(withClientErrorCode: .couldNotPersistAuthInfo)
+        }
     }
 
-    /**
-     * Clears the authentication information for the user with the given id.
-     *
-     * - parameters:
-     *     - storage: The `Storage` which should be cleared of any authentication information.
+    /*
+     * Helper function to update the activeUserAuthInfo and activeUser and persist the changes
      */
-    internal func clearUser(storage: Storage, withUserId userId: String?) throws {
-        objc_sync_enter(authStateLock)
-        defer { objc_sync_exit(authStateLock) }
-
-        // Unwrap the userId and set it to the given id or that of the active user
-        guard let userId = userId != nil ? userId : activeUserAuthInfo?.userID else {
+    internal func updateActiveAuthInfo(withNewAuthInfo authInfo: AuthInfo?) throws {
+        // If passed in nil --> clear active auth state
+        guard let authInfo = authInfo else {
+            authStateHolder.clearState()
```

Comment:
[naming nit] if we're going to keep `authStateHolder`, let's call it `activeAuthStateHolder`